services/navigation/builtin/builtin.go

```diff
@@ -324,9 +320,6 @@ func (svc *builtIn) Reconfigure(ctx context.Context, deps resource.Dependencies,
 		obstacleDetectorNamePairs = append(obstacleDetectorNamePairs, motion.ObstacleDetectorName{
 			VisionServiceName: visionSvc.Name(), CameraName: camera.Name(),
 		})
-		obstacleDetectors = append(obstacleDetectors, &ObstacleDetector{
```

Comment:
[no longer needed as we are transferring the obstacleDetectorNamePairs via extra until motionCfg is available in the Move interface.